oauth-keycloak-authorizer/src/main/java/io/strimzi/kafka/oauth/server/authorizer/UserSpec.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017-2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.server.authorizer;
+
+public class UserSpec {
+
+    private final String type;
+    private final String name;
+
+    private UserSpec(String type, String name) {
+        this.type = type;
+        this.name = name;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+
+    public static UserSpec of(String principal) {
+        String[] spec = principal.split(":");
```

Comment:
Maybe use limit, just in case the name includes `:` (though I'm sure you'll tell me that's impossible)?

Reply:
That's a good catch. We could play super safe and not assume anything about usernames ...